week05/errors_test.go

```diff
@@ -0,0 +1,176 @@
+package demo
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestErrTableNotFoundError(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	exp := "table not found " + tn
+	act := testError.Error()
+
+	if exp != act {
+		t.Fatalf("expected %s, got %s", exp, act)
+	}
+}
+
+func TestErrTableNotFound(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	exp := tn
+	act := testError.TableNotFound()
+
+	if exp != act {
+		t.Fatalf("expected %s, got %s", exp, act)
+	}
+}
+
+func TestIsErrTableNotFound(t *testing.T) {
+	t.Parallel()
+	tn := "testtable"
+
+	testError := ErrTableNotFound{
+		table:		tn,
+	}
+
+	if !IsErrTableNotFound(testError) {
+		t.Fatalf("error asserting type")
+	}
+}
+
+
+func TestErrNoRowsError(t *testing.T) {
+	t.Parallel()
+
+	tn := "testtable"
+	testError := errNoRows{
+		clauses:	Clauses{"id": 1, "name": "John"},
+		table:		tn,
+	}
+
+	exp := "[" + tn + "] no rows found\nquery: " + testError.clauses.String()
+	act := testError.Error()
```

Comment:
Never assert on an error message.

Also, please view the `fmt.Sprintf` function in the standard library to build complex strings.

Reply:
Here as well, I am testing a function that returns a string. I did put in the Sprintf, though, should have been there from the start indeed.

Reply:
Here as well, I am testing a function that returns a string. I did put in the Sprintf, though, should have been there from the start indeed.